python/ray/rllib/ddpg/models.py

```diff
@@ -0,0 +1,241 @@
+
+        self._setup_actor_network(obs_space, ac_space)
+        self._setup_critic_network(obs_space, ac_space)
+        self._setup_critic_loss(ac_space)
+
+        with tf.variable_scope("critic"):
+            self.critic_var_list = tf.get_collection(
+                                     tf.GraphKeys.TRAINABLE_VARIABLES,
+                                     tf.get_variable_scope().name
+                                   )
+            self.critic_vars = TensorFlowVariables(self.critic_loss,
+                                                   self.sess)
+
+        with tf.variable_scope("actor"):
+            self.actor_var_list = tf.get_collection(
+                                    tf.GraphKeys.TRAINABLE_VARIABLES,
+                                    tf.get_variable_scope().name
+                                  )
+            self.actor_vars = TensorFlowVariables(self.output_action,
+                                                  self.sess)
+
+        if (self.config["noise_add"]):
+            params = self.config["noise_parameters"]
+            self.rand_process = OrnsteinUhlenbeckProcess(size=self.ac_size,
+                                                         theta=params["theta"],
+                                                         mu=params["mu"],
+                                                         sigma=params["sigma"])
+            self.epsilon = 1.0
+
+    def _setup_critic_loss(self, action_space):
+        """Sets up critic loss."""
+        self.target_Q = tf.placeholder(tf.float32, [None, 1], name="target_q")
+
+        # compare critic eval to critic_target (squared loss)
+        self.reward = tf.placeholder(tf.float32, [None], name="reward")
+        self.critic_target = tf.expand_dims(self.reward, 1) + \
+            self.config['gamma'] * self.target_Q
+        self.critic_loss = tf.reduce_mean(tf.square(
+                          self.critic_target - self.critic_eval))
+
+    def _setup_critic_network(self, obs_space, ac_space):
+        """Sets up Q network."""
+        with tf.variable_scope("critic", reuse=tf.AUTO_REUSE):
+            self.critic_network = DDPGCritic((self.obs, self.act), 1, {})
+            self.critic_eval = self.critic_network.outputs
+
+        with tf.variable_scope("critic", reuse=True):
+            self.cn_for_loss = DDPGCritic(
+                              (self.obs, self.output_action), 1, {}).outputs
+
+    def _setup_actor_network(self, obs_space, ac_space):
+        """Sets up actor network."""
+        with tf.variable_scope("actor", reuse=tf.AUTO_REUSE):
+            self.actor_network = DDPGActor(
+                                self.obs, self.ac_size,
+                                options={"action_bound": self.action_bound})
+            self.output_action = self.actor_network.outputs
+
+    def get_weights(self):
+        """Returns critic weights, actor weights."""
+        return self.critic_vars.get_weights(), self.actor_vars.get_weights()
```

Comment:
nit: it would be great if we could just have this as one return dict

Reply:
Correct me if I'm wrong, but I think that would make setting weights more annoying?

Reply:
That's true but only slightly; the benefit of doing this is to that different optimizers can use the DDPG evaluator (which is one of the main points of RLlib)

Reply:
OK let's leave this as is and fix it in a later PR if necessary.